src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/LocalDB.Windows.cs

```diff
@@ -0,0 +1,226 @@
+                if (dllPath == null)
+                {
+                    SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.INVALID_PROV, 0, MapLocalDBErrorStateToCode(registryQueryErrorState), string.Empty);
+                    return false;
+                }
+
+                // In case the registry had an empty path for dll
+                if (string.IsNullOrWhiteSpace(dllPath))
+                {
+                    SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.INVALID_PROV, 0, SNICommon.LocalDBInvalidSqlUserInstanceDllPath, string.Empty);
+                    return false;
+                }
+
+                // Load the dll
+                SafeLibraryHandle libraryHandle = Interop.Kernel32.LoadLibraryExW(dllPath.Trim(), IntPtr.Zero, 0);
+
+                if (libraryHandle.IsInvalid)
+                {
+                    SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.INVALID_PROV, 0, SNICommon.LocalDBFailedToLoadDll, string.Empty);
+                    libraryHandle.Dispose();
+                    return false;
+                }
+
+                // Load the procs from the DLLs
+                _startInstanceHandle = Interop.Kernel32.GetProcAddress(libraryHandle, ProcLocalDBStartInstance);
+
+                // Set the delegate the invoke.
+                localDBStartInsanceFunc = (LocalDBStartInstance)Marshal.GetDelegateForFunctionPointer(_startInstanceHandle, typeof(LocalDBStartInstance));
+
+                if (_startInstanceHandle == IntPtr.Zero)
+                {
+                    SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.INVALID_PROV, 0, SNICommon.LocalDBBadRuntime, string.Empty);
+                    libraryHandle.Dispose();
+                    return false;
+                }
+
+                _sqlUserInstanceLibraryHandle = libraryHandle;
+
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Retireves the part of the sqlUserInstance.dll from the registry
+        /// </summary>
+        /// <param name="errorState">In case the dll path is not found, the error is set here.</param>
+        /// <returns></returns>
+        private string GetUserInstanceDllPath(out LocalDBErrorState errorState)
+        {
+            string dllPath = null;
+            using (RegistryKey key = Registry.LocalMachine.OpenSubKey(LocalDBInstalledVersionRegistryKey))
+            {
+                if (key == null)
+                {
+                    errorState = LocalDBErrorState.NO_INSTALLATION;
+                    return null;
+                }
+
+                Version zeroVersion = new Version();
+
+                Version latestVersion = zeroVersion;
+
+                foreach (string subKey in key.GetSubKeyNames())
+                {
+                    try
+                    {
+                        Version currentKeyVersion = new Version(subKey);
```

Comment:
Maybe call `Version.TryParse()` to avoid exceptions? https://msdn.microsoft.com/en-us/library/system.version.tryparse(v=vs.110).aspx